src/main/java/com/conveyal/gtfs/loader/Table.java

```diff
@@ -400,10 +426,48 @@ public String generateUpdateSql (String namespace, int id) {
     }
 
     /**
-     * Generate select all SQL string.
+     * Generate select all SQL string. The minimum requirement parameter is used to determine which fields ought to be
+     * included in the select statement. For example, if "OPTIONAL" is passed in, both optional and required fields
+     * are included in the select. If "EDITOR" is the minimum requirement, editor, optional, and required fields will
+     * all be included.
+     */
+    public String generateSelectSql (String namespace, Requirement minimumRequirement) {
```

Comment:
Via this method I think I understand what requirementsForSpecFields and requirementsForEditorFields mean. It appears that these and editorFields(), requiredFields(), and specFields() are fragmented implementations of an unstated underlying idea: there is a total order on the set of requirement enum values, (i.e. that the <= operator applies to all pairs of elements in that set), as well as an operation to extract the subset of elements less than or equal to a supplied element. If so, you have introduced the idea that the `Requirement` enum values are "levels" like log levels or Maven build cycle targets. If that's the case it should be spelled out in the comments, and reflected in the data structures and operators (methods) defined on those data types.

Concretely, the enum values should be placed in an ordered list or assigned order constants, which allow a generic filtering predicate that returns `true` for elements "up to and including" a supplied level.

Reply:
@abyrd, yes, generally that is my impression of how these requirements might ought to operate.  However, I'm not certain how all of the requirement types might be hierarchically arranged.  For now, I think we should capture your commentary on this in a `TODO` and have further conversation about where the other types (e.g., `PROPRIETARY`, `EXTENSION`, `UNKOWN`) might fit in here.

Reply:
Honestly I don't think there's a clear order to the requirement types. From the existing methods I got the impression you had an implicit ordering in mind, but if not then all the better, let's keep using clearly defined sets of requirement fields for each situation. In that case I just think those should be defined as EnumSets or some other appropriate abstraction instead of the current approach.